lib/rules/require-explicit-slots.js

```diff
@@ -8,8 +8,31 @@ const utils = require('../utils')
 
 /**
  * @typedef {import('@typescript-eslint/types').TSESTree.TypeNode} TypeNode
+ * @typedef {import('@typescript-eslint/types').TSESTree.TypeElement} TypeElement
  */
 
+/**
+ * @param {TypeElement} node
+ * @return {string | null}
+ */
+function getSlotsName(node) {
+  if (
+    node.type === 'TSMethodSignature' ||
+    node.type === 'TSPropertySignature'
+  ) {
+    const key = node.key
+    if (key.type === 'Literal') {
+      return typeof key.value === 'string' ? key.value : null
+    }
+
+    if (key.type === 'Identifier') {
+      return key.name
+    }
+  }
+
+  return null
```

Comment:
Please return early here:

```suggestion
  if (
    node.type !== 'TSMethodSignature' &&
    node.type !== 'TSPropertySignature'
  ) {
    return null
  }

  const key = node.key
  if (key.type === 'Literal') {
    return typeof key.value === 'string' ? key.value : null
  }

  if (key.type === 'Identifier') {
    return key.name
  }

  return null
```